src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -230,89 +174,76 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
-        if (parsed_struct.subnet_states(j).operation_type() == OperationType::INFO) {
-          if (current_SubnetConfiguration.id() ==
-              current_PortConfiguration.fixed_ips(0).subnet_id()) {
-            found_network_type = current_SubnetConfiguration.network_type();
-            found_tunnel_id = current_SubnetConfiguration.tunnel_id();
-            if (!aca_validate_tunnel_id(found_tunnel_id, found_network_type)) {
-              throw std::invalid_argument("found_tunnel_id is invalid");
-            }
-
-            subnet_info_found = true;
-            break;
-          }
-        }
-      }
-
-      if (!subnet_info_found) {
-        ACA_LOG_ERROR("Not able to find the info for port with subnet ID: %s.\n",
-                      current_PortConfiguration.fixed_ips(0).subnet_id().c_str());
-        overall_rc = -EXIT_FAILURE;
-      } else {
-        overall_rc = EXIT_SUCCESS;
-      }
+      port_cidr = virtual_ip_address + "/" + found_prefix_len;
 
-      if (overall_rc == EXIT_SUCCESS) {
-        ACA_LOG_DEBUG("Port Operation:%s: project_id:%s, vpc_id:%s, network_type:%d, virtual_ip_address:%s, "
-                      "virtual_mac_address:%s, neighbor_host_ip_address:%s, tunnel_id:%d\n ",
-                      aca_get_operation_string(current_PortState.operation_type()),
-                      current_PortConfiguration.project_id().c_str(),
-                      current_PortConfiguration.vpc_id().c_str(), found_network_type,
-                      virtual_ip_address.c_str(), virtual_mac_address.c_str(),
-                      host_ip_address.c_str(), found_tunnel_id);
-
-        overall_rc = ACA_OVS_L2_Programmer::get_instance().create_update_neighbor_port(
-                current_PortConfiguration.vpc_id(),
-                current_PortConfiguration.network_type(), host_ip_address,
-                found_tunnel_id, culminative_dataplane_programming_time);
+      ACA_LOG_DEBUG("Port Operation: %s: port_id: %s, project_id:%s, vpc_id:%s, network_type:%d, "
+                    "virtual_ip_address:%s, virtual_mac_address:%s, generated_port_name: %s, port_cidr: %s, tunnel_id: %d\n",
+                    aca_get_operation_string(current_PortState.operation_type()),
+                    current_PortConfiguration.id().c_str(),
+                    current_PortConfiguration.project_id().c_str(),
+                    current_PortConfiguration.vpc_id().c_str(), found_network_type,
+                    virtual_ip_address.c_str(), virtual_mac_address.c_str(),
+                    generated_port_name.c_str(), port_cidr.c_str(), found_tunnel_id);
+
+      overall_rc = ACA_OVS_L2_Programmer::get_instance().create_port(
+              current_PortConfiguration.vpc_id(), generated_port_name, port_cidr,
+              found_tunnel_id, culminative_dataplane_programming_time);
+
+      break;
+
+    case OperationType::UPDATE:
+      // only delete scenario is supported now
+      // VM was created with port specified, then delete the VM
```

Comment:
We could discuss this option: In the VM update scenario, when PM receives the update request and determine it is a delete for the host, PM could issue a DEL. This could simplify ACA implementation. What do you think of?

Reply:
That's a good idea, keep in mind PM needs to determine that anyway because it needs to send down a neighbor table for the corresponding neighbor delete (same logic as port create and send down neighbor table).

Reply:
agreed. Let us use the same issue to track the PM change then.

Reply:
port delete scenario tracking issue has been updated: https://github.com/futurewei-cloud/alcor/issues/103